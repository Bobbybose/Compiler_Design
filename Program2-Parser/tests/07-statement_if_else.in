if(a) print(b); else print(c);
if(a) if(c) d; else e;
