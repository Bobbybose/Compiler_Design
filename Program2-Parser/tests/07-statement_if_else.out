statementList: 1,1
|   
+-- statementList (EMPTY STRING): 1,1
|   
+-- statement: 1,1
|   |   
|   +-- if: 1,1
|   |   
|   +-- (: 1,3
|   |   
|   +-- expression: 1,4
|   |   |   
|   |   +-- a: 1,4
|   |   |   
|   |   *   
|   |   
|   +-- ): 1,5
|   |   
|   +-- statement: 1,7
|   |   |   
|   |   +-- expression: 1,7
|   |   |   |   
|   |   |   +-- function_call: 1,7
|   |   |   |   |   
|   |   |   |   +-- print: 1,7
|   |   |   |   |   
|   |   |   |   +-- (: 1,12
|   |   |   |   |   
|   |   |   |   +-- function_args: 1,13
|   |   |   |   |   |   
|   |   |   |   |   +-- expression: 1,13
|   |   |   |   |   |   |   
|   |   |   |   |   |   +-- b: 1,13
|   |   |   |   |   |   |   
|   |   |   |   |   |   *   
|   |   |   |   |   |   
|   |   |   |   |   *   
|   |   |   |   |   
|   |   |   |   +-- ): 1,14
|   |   |   |   |   
|   |   |   |   *   
|   |   |   |   
|   |   |   *   
|   |   |   
|   |   +-- ;: 1,15
|   |   |   
|   |   *   
|   |   
|   +-- else: 1,17
|   |   
|   +-- statement: 1,22
|   |   |   
|   |   +-- expression: 1,22
|   |   |   |   
|   |   |   +-- function_call: 1,22
|   |   |   |   |   
|   |   |   |   +-- print: 1,22
|   |   |   |   |   
|   |   |   |   +-- (: 1,27
|   |   |   |   |   
|   |   |   |   +-- function_args: 1,28
|   |   |   |   |   |   
|   |   |   |   |   +-- expression: 1,28
|   |   |   |   |   |   |   
|   |   |   |   |   |   +-- c: 1,28
|   |   |   |   |   |   |   
|   |   |   |   |   |   *   
|   |   |   |   |   |   
|   |   |   |   |   *   
|   |   |   |   |   
|   |   |   |   +-- ): 1,29
|   |   |   |   |   
|   |   |   |   *   
|   |   |   |   
|   |   |   *   
|   |   |   
|   |   +-- ;: 1,30
|   |   |   
|   |   *   
|   |   
|   *   
|   
*   
statementList: 2,2
|   
+-- statementList (EMPTY STRING): 2,2
|   
+-- statement: 2,1
|   |   
|   +-- if: 2,1
|   |   
|   +-- (: 2,3
|   |   
|   +-- expression: 2,4
|   |   |   
|   |   +-- a: 2,4
|   |   |   
|   |   *   
|   |   
|   +-- ): 2,5
|   |   
|   +-- statement: 2,7
|   |   |   
|   |   +-- if: 2,7
|   |   |   
|   |   +-- (: 2,9
|   |   |   
|   |   +-- expression: 2,10
|   |   |   |   
|   |   |   +-- c: 2,10
|   |   |   |   
|   |   |   *   
|   |   |   
|   |   +-- ): 2,11
|   |   |   
|   |   +-- statement: 2,13
|   |   |   |   
|   |   |   +-- expression: 2,13
|   |   |   |   |   
|   |   |   |   +-- d: 2,13
|   |   |   |   |   
|   |   |   |   *   
|   |   |   |   
|   |   |   +-- ;: 2,14
|   |   |   |   
|   |   |   *   
|   |   |   
|   |   +-- else: 2,16
|   |   |   
|   |   +-- statement: 2,21
|   |   |   |   
|   |   |   +-- expression: 2,21
|   |   |   |   |   
|   |   |   |   +-- e: 2,21
|   |   |   |   |   
|   |   |   |   *   
|   |   |   |   
|   |   |   +-- ;: 2,22
|   |   |   |   
|   |   |   *   
|   |   |   
|   |   *   
|   |   
|   *   
|   
*   
